### PR TITLE
Use std::call_once to get rid of compile-time warning.

### PR DESCRIPTION
**Pull request by whchung**
This change is potentially big so I'm asking you guys to help review it.

### CODE REVIEW

**Review by sjw36 — APPROVED** [review]
Nice!

File: mlir/tools/mlir-miopen-lib/mlir-miopen-lib.cpp [hunk 1]
```diff
@@ -34,10 +34,11 @@ struct MiirHandle_s {
       return registry;
     };
     static MLIRContext context(getRegistry());
-    static bool once = []() {
+    static std::once_flag once;
+    std::call_once(once, []() {
       context.loadDialect<miopen::MIOpenDialect, StandardOpsDialect>();
       return true;
```
Comment (jerryyin) [inline mlir/tools/mlir-miopen-lib/mlir-miopen-lib.cpp]: Looks like the return is useless now so better get rid of it.

File: mlir/tools/mlir-miopen-lib/mlir-miopen-lib.cpp [hunk 2]
```diff
@@ -62,7 +63,8 @@ struct MiirHandle_s {
 // condition of one thread doing intialization and another doing
 // lowering.
 bool miirLazyInit() {
-  static const bool once = []() {
+  static bool once;
+  std::call_once(once, []() {
```
Comment (jerryyin) [inline mlir/tools/mlir-miopen-lib/mlir-miopen-lib.cpp]: Same here:
 - Use std::once_flag
 - get rid of return

File: mlir/tools/mlir-miopen-lib/mlir-miopen-lib.cpp (hunk 1 above)
Comment (whchung) [reply mlir/tools/mlir-miopen-lib/mlir-miopen-lib.cpp]: done

File: mlir/tools/mlir-miopen-lib/mlir-miopen-lib.cpp (hunk 2 above)
Comment (whchung) [reply mlir/tools/mlir-miopen-lib/mlir-miopen-lib.cpp]: done

**Review by jerryyin — APPROVED** [review]
LGTM

**Pull request merged**